### PR TITLE
Migrate <Layout /> react router v5 routes to react router v6

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #33834 

This PR migrates routes mounted inside the `<Layout />` component to V6.

## Progress

- [x] Migrated all /routes.tsx except repo
- [x] Started migrating all enterprise routes
- [x] repo route 💀

## Test plan

- Almost every route was manually tested (only repeating simple routes without params weren't).
- CI

## App preview:

- [Web](https://sg-web-ps-migrate-layout-routes.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.18 kb) | 0.00% (+0.18 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/migrate-layout-routes/compare-report.html;tab=live_object) for a full comparison between the commits 84bcfbf and 728ea45 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/Layout.tsx [hunk 1]
```diff
@@ -252,23 +257,20 @@ export const Layout: React.FunctionComponent<React.PropsWithChildren<LayoutProps
...
-                                )
-                        )}
-                    </Switch>
+                    <AppRouterContainer>
+                        <Routes>
+                            {props.routes.map(
+                                ({ condition = () => true, ...route }) =>
+                                    condition(context) && (
+                                        <Route
+                                            key="hardcoded-key" // see https://github.com/ReactTraining/react-router/issues/4578#issuecomment-334489490
```
Comment (vovakulikov) [inline client/web/src/Layout.tsx:266]: Do we still need to have this key hardcoded-key workaround, or v6 solves this problem?

Comment (philipp-spiess) [reply client/web/src/Layout.tsx:266]: I think this is unrelated to React router and has to do with how we create this custom routes that we `.map` over. If we don't add the same `key` to all `<Route/>` elements, any route transition will cause all components to be recreated.

This is going to be fixed when work on the flat routes (#46570) and introduce the layout routes which will only unmount when we need them to.

**Comment by philipp-spiess** [comment]
I’m not sure what's going on with the Percy diff but the screens do not show an active state on the "code search" button in prod either? https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/compare 

If anything, the screenshot now match prod which is good? 😅

**Review by valerybugakov — APPROVED** [review]
Nice!

**Pull request merged**